test/Npgsql.Tests/ConnectionStringBuilderTests.cs

```diff
@@ -31,6 +31,38 @@ namespace Npgsql.Tests
 {
     class ConnectionStringBuilderTests
     {
+        [Test, Description("When creating a builder, if ReplicationMode is not None, Pooling should be disabled")]
+        public void ConstructorWhenReplicationModeIsNotNoneShouldDisablePooling()
```

Comment:
For now, I made a force switch whenever logical replication is selected in the connection string. When `ReplicationMode` is `Logical`, we just force `Pooling` to `false`. If we change it to `true` while having `ReplicationMode` = `Logical`, the property set just ignores that and maintain it as `false`.